### ISSUE TITLE
ITCSS Question - Distinguishing between Generic and Element layers

### ISSUE THREAD

**Issue by WebDevLuke** [csswizardry/ama#32]
Hi Harry, big fan of your ITCSS approach. I've used it myself in a number of projects now. One thing that I've struggled with though is how to distinguish between styles that belong in the Generic layer and those which should sit in the Elements layer.

Quoting your Net Magazine article:-
```
03. Generic

The Generic layer is the first one that actually produces any CSS. It houses very high-level, far reaching styles. This layer is seldom modified, and is usually the same across any projects you work on. It contains things like Normalize.css, global box-sizing rules, CSS resets and so on. The Generic layer affects a lot of the DOM, hence it being nice and wide in the Triangle model, and occurring very early on.

04. Elements

These are bare, unclassed HTML elements. What does an h1 look like without a class on it? What does an a look like without a class on it? The Elements layer binds onto bare HTML element (or 'type') selectors only. It is slightly more explicit than the previous layer in that we are now saying 'make every h1 this big', or 'make every a be a certain colour'. It is still a very low-specificity layer, but affects slightly less of the DOM, and is slightly more opinionated, hence its location in the Triangle.

The Elements layer is typically the last one in which we'd find bare, element-based selectors, and is very rarely added to or changed after initial setup. Once we have defined element-level styles, all additions and deviations should be implemented using classes.
```

So I can understand how style declarations which remove padding and margins from many elements are far-reaching enough to sit comfortably in the generic layer, but in inuitcss I've seen for example styles such as the below:-

```
table {
	border-collapse: collapse;
	border-spacing: 0;
}
```

Wondering why this is considered generic enough to sit in that layer when it potentially doesn't affect any more or less of the DOM than the inuitcss table styles in the elements layer?

Basically, I'm just looking for a little more clarity on what should go in which layer. :)

Thanks!

**Comment by csswizardry** [comment]
Hey!

Basically the `border-(collapse|spacing)` rules used to be found in more traditional resets and Normalize.css, so that’s why I included it there.

I believe/guess that Normalize.css will have removed those rules because they are somewhat opinionated, so I guess they share the same sentiment as you.

I keep it in (or moved it back to) the Generic layer because I can’t think of any project that hasn’t called for them, so for me they fit the purposes of a reset-like style.

This is the beauty of ITCSS though… because it isn’t a library or download, it lets developers be very flexible, meaning that a lot of the time there is no right or wrong—you could validly place those rules in the Generic or Elements layer based purely on your own reasoning.
